### DISCUSSION TITLE
Code-base Structure

### DISCUSSION THREAD

**Discussion by IhabBendidi** [pinellolab/DNA-Diffusion#18] [General]
This is the start of the discussion on the general code base structure. What was discussed/done so far was : 

- The code would be built with Pytorch Lightning
- @mateibejan1 built an initial squeleton of the code, that would be filled slowly, in the branch code-base
- We hope to minimize the amount of redundant files that contains classes that contain the same code (For example, we will create only one metrics.py file, instead of many with a metric each one)
- A dataloader of the current dataset is currently built 
- Logging would be done in Weight and Biases, we will create a logging module
- Code will be adapted for distributed gpu training. Anyone without access to a multi-gpu cluster can notify from me @IhabBendidi to run his code on my cluster

Feel free to discuss on any of these points, or add new points

**Comment by IhabBendidi** [comment]
#19 is currently a discussion going on to finalize the code structure and folder organization
